api/src/main/java/org/apache/iceberg/transforms/Identity.java

```diff
@@ -38,6 +38,9 @@ class Identity<T> implements Transform<T, T> {
    */
   @Deprecated
   public static <I> Identity<I> get(Type type) {
+    Preconditions.checkArgument(
+        type.typeId() != Type.TypeID.VARIANT, "Unsupported type for identity: %s", type);
```

Comment:
I think we also need to make sure that `canTransform` doesn't accept variants. We could special case it, but I'm actually wondering if it makes sense for variant to be a primitive type because it is only primitive sometimes.

Reply:
We do only need to special case it if we make Variant not a primitive. I'm not sure what primitive really means in this context though. I think anything that may or may not be scalar is probably not a primitive?